xc/device.ml

```diff
@@ -1409,29 +1409,42 @@ module Vkbd = struct
 end
 
 module Vusb = struct
-  let call_usb_reset hostbus hostport =
-    let usb_reset_script = "/opt/xensource/libexec/usb_reset.py" in
+  let cleanup domid =
     try
-      info "Reset USB device with hostbus %s and hostport %s before passthrough to vm." hostbus hostport;
-      ignore (Forkhelpers.execute_command_get_output usb_reset_script [ hostbus ^ "-" ^ hostport ])
+      let stdout, stderr = Forkhelpers.execute_command_get_output Device_common.usb_reset_script ["cleanup"; "-d"; string_of_int domid] in
+      debug "Got %s %s" stdout stderr
     with _ ->
-      error "Failed to reset USB device with hostbus %s and hostport %s." hostbus hostport;
+      warn "Failed to clean up VM %s" (string_of_int domid)
+
+  let call_usb_reset hostbus hostport domid pid action priv =
+    try
+      match action, priv with
+      | "attach", false ->
+        let stdout, stderr = Forkhelpers.execute_command_get_output Device_common.usb_reset_script [ "attach"; hostbus ^ "-" ^ hostport; "-d"; string_of_int domid; "-p"; string_of_int pid] in
+        debug "Got %s %s" stdout stderr
```

Comment:
These debug lines make no sense when mixed in with other lines in the log because they don't provide enough context what this is about.

Reply:
updated